alerter.c

```diff
@@ -1,31 +1,84 @@
 #include <stdio.h>
 #include <assert.h>
 
+
+#define PRODUCTION_ENV  1
+#define TESTCODE_ENV         2
+
 int alertFailureCount = 0;
+int Current_Env = PRODUCTION_ENV;
+
+/*Conversion from farenheit to celcius*/
+float convertfarenheittocelcius(float farenheit)
+{
+    float celcius = (farenheit - 32) * 5 / 9;
+    return celcius;
+}
 
 int networkAlertStub(float celcius) {
     printf("ALERT: Temperature is %.1f celcius.\n", celcius);
-    // Return 200 for ok
-    // Return 500 for not-ok
-    // stub always succeeds and returns 200
-    return 200;
+    if(celcius<=200)
+    {
+     return 200;
+     }
+    else
+    {
+     return 500;
+    }
 }
 
-void alertInCelcius(float farenheit) {
-    float celcius = (farenheit - 32) * 5 / 9;
-    int returnCode = networkAlertStub(celcius);
+int networkAlertProduction(float celcius){
+  if(celcius<=200)
+  {
+     return 200;
+  }
+  else
+  {
+    return 500;
+  }
+}  
+
+
+/*Selection of the environment: Productio/Testing */
+float SelectAlertEnvironment(float celcius, int Current_Env)
+{
+   float calculated_Value;
+   if(Current_Env == PRODUCTION_ENV )

```

Comment:
Using conditional code for separating production and test code may get complicated as the code size grows.
it is better to use interfaces or functions pointers or lambda functions (if available) to inject either required production or test code based on the envt.
Production code should be simple in the sense, given an input the output should be deterministic, may it be dev environment or test or field